tests/tensor_search/integ_tests/test_dict_score_modifiers.py

```diff
@@ -0,0 +1,573 @@
+
+                # Assert that the first result has _id "6" and 0.8 <= score <= 1.2
+                self.assertIn(res["hits"][0]["_id"], ["1", "7"])
+                self.assertTrue(0.8 <= score_of_first_result <= 1.2)
+
+    # Test combined add to score and multiply score by
+    def test_combined_map_score_modifier(self):
+        """
+        Test that combining adding to score and multiplying score by works for a map score modifier.
+        """
+        for index in [self.structured_default_text_index, self.unstructured_default_text_index]:
+            with self.subTest(index=index.type):
+                # Add documents
+                res = tensor_search.add_documents(
+                    config=self.config,
+                    add_docs_params=AddDocsParams(
+                        index_name=index.name,
+                        docs=[
+                            {"_id": "1", "text_field": "a photo of a cat", "map_score_mods": {"a": 0.5}},
+                            {"_id": "2", "text_field": "a photo of a dog", "map_score_mods": {"b": 0.5}},
+                            {"_id": "3", "text_field": "a photo of a cat", "map_score_mods": {"c": 0.5}},
+                            {"_id": "4", "text_field": "a photo of a cat", "map_score_mods_int": {"a": 1}},
+                            {"_id": "5", "text_field": "a photo of a cat", "map_score_mods_int": {"b": 1}},
+                            {"_id": "6", "text_field": "a photo of a cat", "map_score_mods_int": {"c": 1}},
+                            {"_id": "7", "text_field": "a photo of a cat", "map_score_mods_int": {"c": 1},
+                             "map_score_mods": {"a": 0.5}},
+                        ],
+                        tensor_fields=["text_field"] if isinstance(index, UnstructuredMarqoIndex) else None,
+                    )
+                )
+                # Search with score modifier
+                # 0.5 * 0.5 * 4 = 1 (1 and 7) multiply_score_by
+                # for 7, 0.5 * 0.5 * 4 + 1 * 2 = 3 (6 and 7) add_to_score
+                score_modifier = ScoreModifier(**{
+                    "add_to_score": [{"field_name": "map_score_mods_int.c", "weight": 2}],
+                    "multiply_score_by": [{"field_name": "map_score_mods.a", "weight": 4}]
+                }
+                                               )
+                res = tensor_search.search(
+                    index_name=index.name, config=self.config, text="",
+                    score_modifiers=score_modifier,
+                    result_count=10
+                )
+
+                # Get the score of the first result.
+                score_of_first_result = res["hits"][0]["_score"]
+
+                # Assert that the first result has _id "6" and 5 <= score <= 6
+                self.assertTrue(res["hits"][0]["_id"] == "7")
+                self.assertTrue(2.9 <= score_of_first_result <= 3.1)
+
+    def test_partial_document_update(self):
```

Comment:
Minor comment (don't have to change here), but best to fit most tests within current groupings where possible. e.g. this could be added to existing partial update tests class. This way it's easier to find all partial updates tests